Makefile

```diff
@@ -255,3 +255,8 @@ release-minor: check-release bumpversion-minor release
 
 .PHONY: release-major
 release-major: check-release bumpversion-major release
+
+.PHONY: checkdeps
+checkdeps: # Save the currently installed versions of the dependencies as the latest versions 
+	$(eval allow_list='mlblocks|pandas|numpy|psutil')
+	pip freeze | grep -v "sintel-dev/SigPro.git" | grep -E $(allow_list) > $(OUTPUT_PATH)
```

Comment:
Yes, it'd need to get updated. The idea is to only track the main dependencies since they're the most likely to cause issues and should be handling their dependencies correctly. I don't know how we'd be able to read `install_requires` directly, but maybe we could move our requirements to an external requirements file?

Reply:
I guess my intuition is that `install_requires` would have the necessary packages to track and `install_extra` or something like that would have the packages that we don't track (both get installed in a regular setup). Then we can have `requirements.txt` to track `install_requires`. 

I'm not 100% convinced with my idea, but I just wanted to open up the discussion.

@dyuliu what do you think?